sdk/iot/Azure.Iot.Hub.Service/src/API Design.md

```diff
@@ -42,7 +42,129 @@ APIs for getting statistics about devices and modules, as well as service statis
+
+    /// <summary>
+    /// Create multiple devices. A maximum of 100 creations can be done per call, and each device identity must be unique. For larger scale operations, consider using IoT Hub jobs (https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-identity-registry#import-and-export-device-identities).
+    /// </summary>
+    /// <param name="deviceIdentities">The devices to create.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>The result of the bulk operation.</returns>
+    public virtual async Task<Response<BulkRegistryOperationResult>> CreateIdentitiesAsync(IEnumerable<DeviceIdentity> deviceIdentities, CancellationToken cancellationToken = default)
+
+    /// <summary>
+    /// Update multiple devices. A maximum of 100 updates can be done per call, and each operation must be done on a different identity. For larger scale operations, consider using IoT Hub jobs (https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-identity-registry#import-and-export-device-identities).
+    /// </summary>
+    /// <param name="deviceIdentities">The devices to update.</param>
+    /// <param name="force">If true, the devices will be updated even if their ETag is out of date.
+    /// If false, each device will only be updated if its ETag is up to date.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>The result of the bulk operation.</returns>
+    public virtual async Task<Response<BulkRegistryOperationResult>> UpdateIdentiesAsync(IEnumerable<DeviceIdentity> deviceIdentities, bool force, CancellationToken cancellationToken = default)
+    
+    /// <summary>
+    /// Delete multiple devices. A maximum of 100 deletions can be done per call. For larger scale operations, consider using IoT Hub jobs (https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-identity-registry#import-and-export-device-identities).
+    /// </summary>
+    /// <param name="deviceIdentities">The devices to delete.</param>
+    /// <param name="force">If true, the devices will be deleted even if their ETag is out of date.
+    /// If false, each device will only be deleted if its ETag is up to date.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>The result of the bulk deletion.</returns>
+    public virtual async Task<Response<BulkRegistryOperationResult>> DeleteIdentitiesAsync(IEnumerable<DeviceIdentity> deviceIdentities, bool force, CancellationToken cancellationToken = default)
+
+    /// <summary>
+    /// List a set of device twins.
+    /// </summary>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>A pageable set of device twins.</returns>
+    public virtual AsyncPageable<TwinData> GetTwinsAsync(CancellationToken cancellationToken = default)
```

Comment:
As a customer, What flow should I follow to map device identities to twin data? Can we make that easier?

Reply:
It would have to be on DeviceId. Did you have something in mind to do that?

Reply:
For better or for worse, the service maintains the twin and the identity as separate entities. There's no great way for the SDK to combine the two and present them as a single class

Reply:
Lets take this comment post we conclude on the other open one as it may self resolve.